File: src/main/java/space/mori/regenblock/Listener/RegenBlockEventListenerPlayer.java
package space.mori.regenblock.Listener;

import space.mori.regenblock.*;
import org.bukkit.*;
import java.util.*;
import org.bukkit.event.block.*;
import org.bukkit.entity.*;
import org.bukkit.event.*;
import org.bukkit.event.player.*;

public class RegenBlockEventListenerPlayer implements Listener
{
    private RegenBlock plugin;
    private HashMap<String, Location> playerSelectionLeft;
    private HashMap<String, Location> playerSelectionRight;
    private ArrayList<String> playerSelectionStatus;
    private ArrayList<String> playerEditStatus;
    
    public HashMap<String, Location> getPlayerSelectionLeft() {
        return this.playerSelectionLeft;
    }
    
    public HashMap<String, Location> getPlayerSelectionRight() {
        return this.playerSelectionRight;
    }
    
    public ArrayList<String> getPlayerSelectionStatus() {
        return this.playerSelectionStatus;
    }
    
    public ArrayList<String> getPlayerEditStatus() {
        return this.playerEditStatus;
    }
    
    public RegenBlockEventListenerPlayer(final RegenBlock plugin) {
        this.playerSelectionLeft = new HashMap<String, Location>();
        this.playerSelectionRight = new HashMap<String, Location>();
        this.playerSelectionStatus = new ArrayList<String>();
        this.playerEditStatus = new ArrayList<String>();
        this.plugin = plugin;
    }
    
    @EventHandler
    public void onEvent(final PlayerInteractEvent event) {
        if (event.isCancelled()) {
            return;
        }
        if (!this.playerSelectionStatus.contains(event.getPlayer().getName())) {
            return;
        }
        final Player player = event.getPlayer();
        final Location loc = event.getClickedBlock().getLocation();
        final Action action = event.getAction();
        if (action == Action.LEFT_CLICK_BLOCK) {
            final Location locOld = this.playerSelectionLeft.get(player.getName());
            if ((locOld != null && !locOld.equals((Object)loc)) || locOld == null) {
                this.playerSelectionLeft.put(player.getName(), loc);
                this.plugin.getLog().sendPlayerNormal(player, "Left Block: " + loc.getBlockX() + " " + loc.getBlockY() + " " + loc.getBlockZ());
            }
        }
        else if (action == Action.RIGHT_CLICK_BLOCK) {
            final Location locOld = this.playerSelectionRight.get(player.getName());
            if ((locOld != null && !locOld.equals((Object)loc)) || locOld == null) {
                this.playerSelectionRight.put(player.getName(), loc);
                this.plugin.getLog().sendPlayerNormal(player, "Right Block: " + loc.getBlockX() + " " + loc.getBlockY() + " " + loc.getBlockZ());
            }
        }
        event.setCancelled(true);
    }
    
    @EventHandler
    public void onEvent(final PlayerChangedWorldEvent event) {
        final Player player = event.getPlayer();
        this.playerSelectionLeft.remove(player.getName());
        this.playerSelectionRight.remove(player.getName());
        this.playerSelectionStatus.remove(player.getName());
        this.playerEditStatus.remove(player.getName());
    }
    
    @EventHandler
    public void onEvent(final PlayerJoinEvent event) {
        final Player player = event.getPlayer();
        this.playerSelectionLeft.remove(player.getName());
        this.playerSelectionRight.remove(player.getName());
        this.playerSelectionStatus.remove(player.getName());
        this.playerEditStatus.remove(player.getName());
    }
    
    @EventHandler
    public void onEvent(final PlayerQuitEvent event) {
        final Player player = event.getPlayer();
        this.playerSelectionLeft.remove(player.getName());
        this.playerSelectionRight.remove(player.getName());
        this.playerSelectionStatus.remove(player.getName());
        this.playerEditStatus.remove(player.getName());
    }
}

File: src/main/kotlin/space/mori/regenblock/listener/RegenBlockEventListenerPlayer.kt
package space.mori.regenblock.listener

import space.mori.regenblock.*
import org.bukkit.*
import java.util.*
import org.bukkit.event.block.*
import org.bukkit.event.*
import org.bukkit.event.player.*

class RegenBlockEventListenerPlayer(private val plugin: RegenBlock) : Listener {
    val playerSelectionLeft: HashMap<String, Location> = HashMap()
    val playerSelectionRight: HashMap<String, Location> = HashMap()
    val playerSelectionStatus: ArrayList<String> = ArrayList()
    val playerEditStatus: ArrayList<String> = ArrayList()

    @EventHandler
    fun onEvent(event: PlayerInteractEvent) {
        if (event.isCancelled) {
            return
        }
        if (!this.playerSelectionStatus.contains(event.player.name)) {
            return
        }
        val player = event.player
        val loc = event.clickedBlock.location
        val action = event.action
        if (action == Action.LEFT_CLICK_BLOCK) {
            val locOld = this.playerSelectionLeft[player.name]
            if (locOld != null && locOld != loc as Any || locOld == null) {
                this.playerSelectionLeft[player.name] = loc
                this.plugin.log!!.sendPlayerNormal(
                    player,
                    "Left Block: ${loc.blockX} ${loc.blockY} ${loc.blockZ}"
                )
            }
        } else if (action == Action.RIGHT_CLICK_BLOCK) {
            val locOld = this.playerSelectionRight[player.name]
            if (locOld != null && locOld != loc as Any || locOld == null) {
                this.playerSelectionRight[player.name] = loc
                this.plugin.log!!.sendPlayerNormal(
                    player,
                    "Right Block: ${loc.blockX} ${loc.blockY} ${loc.blockZ}"
                )
            }
        }
        event.isCancelled = true
    }

    @EventHandler
    fun onEvent(event: PlayerChangedWorldEvent) {
        val player = event.player
        this.playerSelectionLeft.remove(player.name)
        this.playerSelectionRight.remove(player.name)
        this.playerSelectionStatus.remove(player.name)
        this.playerEditStatus.remove(player.name)
    }

    @EventHandler
    fun onEvent(event: PlayerJoinEvent) {
        val player = event.player
        this.playerSelectionLeft.remove(player.name)
        this.playerSelectionRight.remove(player.name)
        this.playerSelectionStatus.remove(player.name)
        this.playerEditStatus.remove(player.name)
    }

    @EventHandler
    fun onEvent(event: PlayerQuitEvent) {
        val player = event.player
        this.playerSelectionLeft.remove(player.name)
        this.playerSelectionRight.remove(player.name)
        this.playerSelectionStatus.remove(player.name)
        this.playerEditStatus.remove(player.name)
    }
}
